sql/driver/operators_it_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package driver_test
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+	"github.com/hazelcast/hazelcast-go-client/sql/driver"
+)
+
+type employee struct {
+	ID            int `json:"id"`
+	PersonnelInfo `json:"personnel_info"`
+	CompanyInfo   `json:"company_info"`
+}
+type PersonnelInfo struct {
+	Name string `json:"name"`
+	Age  int    `json:"age"`
+}
+type CompanyInfo struct {
+	Title   string `json:"title"`
+	DepCode int    `json:"dep_code"`
+}
+
+func TestNestedJSONQuery(t *testing.T) {
+	it.SkipIf(t, "hz < 5.1")
+	it.SQLTester(t, func(t *testing.T, client *hz.Client, config *hz.Config, m *hz.Map, mapName string) {
+		defer driver.SetSerializationConfig(nil)
+		const rowCount = 50
+		db, err := initDB(t, config)
+		assert.Nil(t, err)
+		defer db.Close()
+		ms := createMappingStr(mapName, "bigint", "json")
+		it.Must(createMapping(t, db, ms))
+		// Insert random employee entries.
+		employees, err := populateMapWithEmployees(m, rowCount)
+		assert.Nil(t, err)
+		// Test query on nested json
+		complexQuery := fmt.Sprintf(`SELECT JSON_VALUE(this, '$.personnel_info.name') FROM "%s" WHERE CAST(JSON_VALUE(this, '$.id') AS TINYINT)=0`, mapName)
+		rows, err := db.QueryContext(context.Background(), complexQuery)
+		assert.Nil(t, err)
+		defer rows.Close()
+		var count int
+		for ; rows.Next(); count++ {
+			var name string
+			it.Must(rows.Scan(&name))
+			assert.Equal(t, employees[0].PersonnelInfo.Name, name)
+		}
+		assert.Equal(t, 1, count)
```

Comment:
How about using `db.QueryRowContext` instead of `QueryContext` since a single row is expected?

Reply:
Thanks! I wasn't aware of that.